scio-core/src/main/java/com/spotify/scio/transforms/BaseAsyncLookupDoFn.java

```diff
@@ -155,40 +159,58 @@ public void startBundle(StartBundleContext context) {
     semaphore.release(maxPendingRequests);
   }
 
+  // kept for binary compatibility. Must not be used
+  // TODO: remove in 0.15.0
+  @Deprecated
```

Comment:
Will this deprecation warning get logged on the user's side? I guess only if they're pulling in an older Beam version that uses the old API, right?

Reply:
Only if extending the class and overriding this method which should never happen.
However a library compiled with the old API will call this method, so I've left it for now.